VENDORING.md

```diff
@@ -0,0 +1,235 @@
+#  version = "2.4.0"
+
+```
+This file is interesting because it contains all the information the developer
+needs to modify it properly.
+
+From this point, you can vendor any package needed from your code.
+
+### Vendor a newly imported package
+
+This case describes a situation where the developer is adding some new code
+relying on a package never imported before. Let's say the new package is
+`github.com/foo/bar`, here is what needs to be done:
+```bash
+$ dep ensure
+$ dep prune
+```
+
+At this point, only the needed files will have been copied into `vendor`.
+
+One more thing we might want here is to ensure the vendored package refers
+to a specific version. This is what is called a constraint with `dep`.
+By modifying `Gopkg.toml`, the developer can decide to force our package
+`github.com/foo/bar` to point to the revision `f5742cb6`. Here is what the
+file should look like:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+```
+
+Or in case he wants to point to a specific version `1.2.4`:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  version = "1.2.4"
+```
+
+Or in case he wants to point to a specific branch `work-dev`:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  branch = "work-dev"
+```
+
+At last, a very powerful constraint is to be able to specify if the source
+of repository comes from a forked repository:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  branch = "fork-work-dev"
+  source = "github.com/myfork/bar"
+```
+
+__Note:__ Constraining a package is important because this is the only way to
+make sure about the version of an imported package. This prevents from any
+build failure that could result from a package change on the upstream.
+
+### Update a vendored package
+
+This case describes a situation where the developer adds some code relying on
+another version of a package already vendored. This is a very simple case since
+the package is already part of the vendoring. The developer will have to modify
+`Gopkg.toml` by updating the vendoring version:
+
+```diff
+--- Gopkg.toml.orig     2018-01-23 09:04:54.160760426 -0800
++++ Gopkg.toml.new      2018-01-23 09:05:09.164094911 -0800
+@@ -1,3 +1,3 @@
+ [[constraint]]
+   name = "github.com/foo/bar"
+-  revision = "f5742cb6"
++  revision = "a4be45c7"
+```
+
+And let `dep` update the vendoring by relying on this new revision:
+```bash
+$ dep ensure
+$ dep prune
+```
+
+### Remove a vendored package
+
+This case describes a situation where the developer modifies the existing
+code, no longer relying on a vendored package. Running the usual commands
+will remove the vendored package from `vendor` directory and from
+`Gopkg.lock`:
+```bash
+$ dep ensure
+$ dep prune
+```
+
+If a constraint has been set in `Gopkg.toml`, this is the responsibility of
+developer to remove it manually from this file.
+
+### Test a pending pull request
+
+This case describes a situation where the developer needs to perform some
```

Comment:
Lines 183-184 suggested rewrite: The following case describes a situation where you need to perform testing on your code due to a pending pull request from a vendored package.

Reply:
done